src/requireAuth.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { Component, Fragment } from 'react';
+import SignUpPage from 'components/SignUpPage';
+
+interface IAuthenticateState {
+  isLogged: boolean
+}
+
+interface IAuthenticateProps {
+  location: { hash: string }
+}
+export default function (ComposedComponent: React.ComponentType): React.ComponentType {
+  return class Authenticate extends Component<IAuthenticateProps, IAuthenticateState> {
+    constructor(props: IAuthenticateProps) {
+      super(props);
```

Comment:
unfortunately, it's not working as expected. I can change token, userID or userName and I will be able to send a message as Dilven :P

Reply:
we should use backed for requests to firebase/websockets, then we could check token on every request. Also we shouldn't store userName and userId on front-end. It's quite easy to change it and publish message as someone else. Imho It should come back from API to App with a message.

Reply:
Totally agree. This is the part I didn't make yet. So I put it in localStorage for now. Normally, it will be only a token.